### PR TITLE
linenoise: 1.0.10 -> 1.0-34-g93b2db9

### PR DESCRIPTION
**Pull request by remexre**
## Description of changes

Updates linenoise, and changes its packaging to be more typical for libraries.

Weird things:

- linenoise hasn't had an official release in years despite ongoing development, so this uses a `git describe` version number
- linenoise doesn't include a build system, so this hardcodes compile and link commands
- linenoise doesn't ship a pkg-config file, so we include one

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/linenoise/default.nix [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
   };
 
-  buildPhase = ./create-pkg-config-file.sh;
+  nativeBuildInputs = [ validatePkgConfig ];
+
+  buildPhase = ''
+    $CC -c -gctf -gdwarf -o linenoise.o linenoise.c
+    $CC -shared -o liblinenoise.so linenoise.o
+    $AR rcs liblinenoise.a linenoise.o
+  '';
```
Comment (Janik-Haag) [inline pkgs/development/libraries/linenoise/default.nix:20]:
```suggestion
  buildPhase = ''
    runHook preBuild

    $CC -c -gctf -gdwarf -o linenoise.o linenoise.c
    $CC -shared -o liblinenoise.so linenoise.o
    $AR rcs liblinenoise.a linenoise.o

    runHook postBuild
  '';
```
please add our hooks

File: pkgs/development/libraries/linenoise/default.nix [hunk 2]
```diff
@@ -1,29 +1,36 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ fetchFromGitHub, lib, stdenv, validatePkgConfig }:
```
Comment (Janik-Haag) [inline pkgs/development/libraries/linenoise/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, validatePkgConfig
}:
```
it's nicer to read

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -13699,6 +13699,12 @@
     githubId = 801525;
     name = "rembo10";
   };
+  remexre = {
+    email = "nathan+nixpkgs@remexre.com";
+    github = "remexre";
+    githubId = 4196789;
+    name = "Nathan Ringo";
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:13707]:
```suggestion
  remexre = {
    email = "nathan+nixpkgs@remexre.com";
    github = "remexre";
    githubId = 4196789;
    name = "Nathan Ringo";
  };
```
please split this out into it's own commit called: `maintainers add remexre`

**Comment by remexre** [comment]
All done.

File: pkgs/development/libraries/linenoise/default.nix [hunk 4]
```diff
@@ -1,29 +1,48 @@
...
+
+    runHook postInstall
   '';
 
   meta = {
     homepage = "https://github.com/antirez/linenoise";
     description = "A minimal, zero-config, BSD licensed, readline replacement";
-    maintainers = with lib.maintainers; [ fstamour ];
+    maintainers = with lib.maintainers; [ fstamour remexre ];
     platforms = lib.platforms.unix;
```
Comment (Janik-Haag) [inline pkgs/development/libraries/linenoise/default.nix:46]: if this is supposed to build on darwin mark darwin as broken since the build currently fails in ci, if it is only supposed to build on linux set platforms to linux and if it's supposed to build on all unix-like platforms except for darwin set `badPlatforms = lib.platforms.darwin;`

**Review by Janik-Haag — APPROVED** [review]

**Comment by Janik-Haag** [comment]
@fstamour would you mind also giving this a review since you are the current package maintainer?

**Review by fstamour — APPROVED** [review]
